pkg/trait/knative_service_test.go

```diff
@@ -512,3 +514,26 @@ func createKnativeServiceTestEnvironment(t *testing.T, trait *traitv1.KnativeSer
 
 	return environment
 }
+func TestServiceAnnotation(t *testing.T) {
+	annotationsTest := map[string]string{"haproxy.router.openshift.io/balance": "true"}
+
+	environment := createTestRouteEnvironment(t, KnativeServiceTestName)
+	environment.Integration.Spec.Traits = v1.Traits{
+		Route: &traitv1.RouteTrait{
```

Comment:
Why are you testing Route annotation? I think you need to test `KnativeServiceTrait` instead.